modules/axhal/src/arch/loongarch64/trap.S

```diff
@@ -0,0 +1,158 @@
+.macro SAVE_REGS
+    st.d    $ra, $sp, 8
+    csrrd   $t0, KSAVE_T0
+    st.d    $t0, $sp, 12*8
+
+    st.d    $a0, $sp, 4*8

```

Comment:
Add macros to load/save general registers, and we can reuse it in `UspaceContext::enter_uspace`, like `PUSH_GENERAL_REGS`/`POP_GENERAL_REGS` in riscv.